### Task 1
Із генеральної сукупності, розподіленої за нормальним законом, отримано вибірку об'ємом n = 20

|Xi| 1 | 2 | 4 | 6 | 7 |
|-:|--:|--:|--:|--:|--:|
|ni| 2 | 3 |10 | 3 | 2 |

Знайти з надійністю у = 0,95 інтервальну оцінку для середнього квадратичного відхилення о генеральної сукупності.


Для розрахунку інтервальної оцінки середнього значення генеральної сукупності нам знадобиться розрахувати виборочне середнє значення:  
$$\overline X =  \frac{1}{N} \sum_{i=1}^{k} x_in_i$$   
А також дисперсія, яка розраховується за формулою:  
$$ D = \frac{\sum_{i=1}^{k} n_i(x_i - \overline X)^2}{N} $$   
i стандартне відхилення:  $\sigma = \sqrt{D}$,   
де k - кількість варіант, N - об'єм вибірки, тобто 20.  
Щоб розрахувати довірчий інтервал з надійністю 95% нам потрібно взяти відхилення $ 1.96*\sigma/\sqrt{N}$ в обидві сторони від  виборочного середнього значення. Коефіціент 1.96 отримуємо з таблиці Лапласа для заданої точності.


In [9]:
import numpy as np
x = np.array([1, 2, 4, 6, 7])
n = np.array([2, 3, 10, 3, 2])
N = sum(n)

# Виборочне середнє
X = sum(x*n)/N 

# Дисперсія
D = sum([n[i]*(x[i] - X)**2 for i in range(len(x))])/N

# Стандартне відхилення
sd = round(D**0.5, 4)

# Довірчий інтервал
x_range = [round(X - 1.96*sd/N**0.5, 2), round(X + 1.96*sd/N**0.5, 2)] 

print(f'Виборочне середнє значення дорівнює: {X}')
print(f"Дисперсія дорівнює: {D}")
print(f'Середнє квадратичне відхилення дорівнює: {sd}') 
print(f'Довірчий інтервал, отриманий на основі виборки: {x_range}')

Виборочне середнє значення дорівнює: 4.0
Дисперсія дорівнює: 3.0
Середнє квадратичне відхилення дорівнює: 1.7321
Довірчий інтервал, отриманий на основі виборки: [3.24, 4.76]


**Висновок:** Середнє значення генеральної сукупності з ймовірністю 95% належить інтервалу [3.24, 4.76].